Clean area/connectors related github issues (#1929)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/applications-wsus-nsclient.md

---
id: applications-wsus-nsclient

> title: Microsoft WSUS Server

Reviewer: see fr comment

```bash
dnf install centreon-pack-applications-wsus-nsclient
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-pack-applications-wsus-nsclient
```

</TabItem>
<TabItem value="Debian 11" label="Debian 11">

```bash
apt install centreon-pack-applications-wsus-nsclient
```

</TabItem>
</Tabs>

Whatever the license type (*online* or *offline*), install the **Microsoft WSUS** Pack through
the **Configuration > Plugin Packs > Manager** menu.

### Plugin

Use the commands below according to your operating system's package manager:

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf install centreon-nrpe3-plugin
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-nrpe3-plugin
```

</TabItem>
<TabItem value="Debian 11" label="Debian 11">

```bash
apt install centreon-nrpe3-plugin
```

</TabItem>
</Tabs>

## Configuration

### Host

* Log into Centreon and add a new host through **Configuration > Hosts**.
* Fill the **Name**, **Alias** & **IP Address/DNS** fields according to your **WSUS Server** server settings.

> * Apply the **App-Wsus-NRPE-custom** template to the host.

Reviewer: see fr comment

| Mandatory   | Macro                     | Description                                                                            |
|:------------|:--------------------------|:---------------------------------------------------------------------------------------|
|             | NSCPRESTAPIEXTRAOPTIONS   | Any extra option you may want to add to every command line (eg. a --verbose flag)      |
|             | NSCPRESTAPILEGACYPASSWORD | API Password                                                                           |
|             | NSCPRESTAPIPORT           | API Port                                                                               |
|             | NSCPRESTAPIPROTO          | API Protocol                                                                           |
|             | WSUSPORT                  | WSUS Server port                                                                       |
|             | WSUSSERVER                | WSUS Server name                                                                       |

</TabItem>
</Tabs>

## How to check in the CLI that the configuration is OK and what are the main options for?

> 

Reviewer: "The following examples come for the RestAPI template" ?

</TabItem>
<TabItem value="Synchronisation-Status" label="Synchronisation-Status">

| Metric Name                   | Unit  |
|:------------------------------|:------|
| synchronisation-progress      |       |
| synchronisation-status        |       |
| last-synchronisation-duration |       |
| last-synchronisation-status   |       |

</TabItem>
<TabItem value="Update-Status" label="Update-Status">

| Metric Name         | Unit  |
|:--------------------|:------|
| needed-by-computers |       |
| needing-files       |       |
| up-to-date          |       |
| with-client-errors  |       |
| with-server-errors  |       |

</TabItem>
</Tabs>

## Prerequisites

### Centreon NSClient++

<Tabs groupId="sync">
<TabItem value="App-Wsus-NRPE-custom" label="App-Wsus-NRPE-custom">

To monitor a *WSUS Server* through NSClient++ API, install the Centreon packaged version
of the NSClient++ agent. Please follow our [official documentation](../getting-started/how-to-guides/centreon-nsclient-tutorial.md)
and make sure that the **NRPE Server** configuration is correct.

</TabItem>
<TabItem value="App-Wsus-NSClient-05-Restapi-custom" label="App-Wsus-NSClient-05-Restapi-custom">

> To monitor an *WSUS Server* through NSClient++ API, install the Centreon packaged version

Reviewer 2: ```suggestion
To monitor a *WSUS Server* through NSClient++ API, install the Centreon packaged version
```

of the NSClient++ agent. Please follow our [official documentation](../getting-started/how-to-guides/centreon-nsclient-tutorial.md)
and make sure that the **Webserver / RESTApi** configuration is correct.

</TabItem>
</Tabs>

## Setup

### Monitoring Pack

If the platform uses an *online* license, you can skip the package installation

> instruction below as it is not required to have the pack displayed within the

Reviewer 2: ```suggestion
instructions below as it is not required in order to have the pack displayed within the
```

    --warning-updates="" \
    --critical-updates="" \
    --warning-approved-updates="" \
    --critical-approved-updates="" \
    --warning-declined-updates="" \
    --critical-declined-updates=""\
    --warning-not-approved-updates="" \
    --critical-declined-updates="" \
    --warning-stale-updates="" \
    --critical-stale-updates="" \
    --warning-expired-updates="" \
    --critical-expired-updates="" \
    --verbose'\
    --use-new-perfdata
```

The expected command output is shown below:

```bash
OK: Computers: 120 Computer Groups: 6 Updates: 19 Approved Updates: 3 Declined Updates: 14 Not Approved Updates: 22 Stale Updates: 1 Expired Updates: 5 | 
```

All available options for a given mode can be displayed by adding the
`--help` parameter to the command:

```bash
/usr/lib/centreon/plugins//centreon_nsclient_restapi.pl \
    --plugin=apps::nsclient::restapi::plugin \
    --mode=query \
    --help
```

All available modes can be displayed by adding the `--list-mode` parameter to
the command:

```bash
/usr/lib/centreon/plugins//centreon_nsclient_restapi.pl \
    --plugin=apps::nsclient::restapi::plugin \
    --list-mode
```

### Troubleshooting

> Please find the [troubleshooting documentation](../getting-started/how-to-guides/troubleshooting-plugins.md)

Reviewer 2: ```suggestion
Please find the [troubleshooting documentation here](../getting-started/how-to-guides/troubleshooting-plugins.md)
```
